internal/exec/stages/files/files.go

```diff
@@ -93,9 +93,12 @@ func (s stage) Run(config types.Config) error {
 // checkRelabeling determines whether relabeling is supported/requested so that
 // we only collect filenames if we need to.
 func (s *stage) checkRelabeling() error {
-	if !distro.SelinuxRelabel() || distro.RestoreconCmd() == "" {
+	if !distro.SelinuxRelabel() {
 		s.Logger.Debug("compiled without relabeling support, skipping")
 		return nil
+	} else if distro.RestoreconCmd() == "" {
+		s.Logger.Debug("targeting distro without restorecon, skipping relabeling")
```

Comment:
Should this error message only be printed if `distro.SelinuxRelabelMode == "systemd"`?